### PR TITLE
Add investigation for STOPSIGNAL instruction using kind and podman on Fedora 35

### PR DESCRIPTION
**Pull request by avisiedo**
- Add Dockerfile to build container for the investigation which contains STOPSIGNAL instruction.
- Script for trap several signals.
- Pod configuration.
- kustomize configuration.
- Makefile for automating the steps.
- Documentation at `README.md` file.

### CODE REVIEW

File: incubator/013-signalstop/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM quay.io/fedora/fedora:35
+COPY demo-signal.sh /demo-signal.sh
+STOPSIGNAL SIGINT
+ENTRYPOINT ["/demo-signal.sh"]
```
Comment (tiran) [inline incubator/013-signalstop/Dockerfile]: * https://docs.docker.com/engine/reference/builder/#entrypoint
* https://docs.docker.com/engine/reference/builder/#cmd

Comment (tiran) [reply incubator/013-signalstop/Dockerfile]:
```suggestion
CMD ["/demo-signal.sh"]
```

Comment (avisiedo) [reply incubator/013-signalstop/Dockerfile]: Thanks! I will exchange the entrypoint by a cmd instruction.

File: incubator/013-signalstop/demo-signal.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+function trap_sigterm { echo "Exiting by SIGTERM"; exit 0; }
+function trap_sigusr1 { echo "Exiting by SIGUSR1"; exit 0; }
+function trap_sigrtmin3 { echo "Exiting by SIGRTMIN+3"; exit 0; }
+
+trap trap_sigint SIGINT
+trap trap_sigterm SIGTERM
+trap trap_sigusr1 SIGUSR1
+trap trap_sigrtmin3 "SIGRTMIN+3"
+
+while true; do sleep 1; done
```
Comment (tiran) [inline incubator/013-signalstop/demo-signal.sh]:
```suggestion
while true; do
    echo -n "."
    sleep 1
done
```

File: incubator/013-signalstop/demo-signal.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+function trap_sigint { echo "Exiting by SIGINT";  exit 0; }
```
Comment (tiran) [inline incubator/013-signalstop/demo-signal.sh]:
```suggestion
function trap_sigint { echo -e "\nExiting by SIGINT" >&2;  exit 0; }
```

File: incubator/013-signalstop/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+#   fieldref:
+#     fieldpath: spec.baseDomain
+# - name: CLUSTER_DOMAIN
+#   objref: 
+#     kind: ConfigMap
+#     name: freeipa
+#     apiVersion: v1
+#   fieldref:
+#     fieldpath: data.CLUSTER_DOMAIN
+
```
Comment (avisiedo) [inline incubator/013-signalstop/kustomization.yaml]: Clean-up this commented section

File: incubator/013-signalstop/demo-signal.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+function trap_sigterm { echo "Exiting by SIGTERM"; exit 0; }
+function trap_sigusr1 { echo "Exiting by SIGUSR1"; exit 0; }
+function trap_sigrtmin3 { echo "Exiting by SIGRTMIN+3"; exit 0; }
```
Comment (avisiedo) [inline incubator/013-signalstop/demo-signal.sh:6]:
```suggestion
function trap_signal {
  local signal="$1"
  echo -e "\nExiting by ${signal}" >&2
  exit 0
}

for signal in SIGINT SIGTERM SIGUSR1 "SIGRTMIN+3"
do
  trap "trap_signal '${signal}'" "${signal}"
done
```

I like the suggestion and I add a bit more for removing code duplication that I included initially.

File: incubator/013-signalstop/demo-signal.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+trap trap_sigint SIGINT
+trap trap_sigterm SIGTERM
+trap trap_sigusr1 SIGUSR1
+trap trap_sigrtmin3 "SIGRTMIN+3"
```
Comment (avisiedo) [inline incubator/013-signalstop/demo-signal.sh]:
```suggestion
function trap_signal {
  local signal="$1"
  echo -e "\nExiting by ${signal}" >&2
  exit 0
}

for signal in SIGINT SIGTERM SIGUSR1 "SIGRTMIN+3"
do
  trap "trap_signal '${signal}'" "${signal}"
done
```

I like the suggestion and I add a bit more changes for removing duplicated code I included at the beginning. Thanks @tiran

**Review by tiran — APPROVED** [review]

**Pull request merged**